src/ip_address.h

```diff
@@ -38,6 +38,11 @@
 #include <afunix.h>
 typedef unsigned short int sa_family_t;
 
+#elif __APPLE__
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <sys/un.h>      /* definitions for UNIX domain sockets */
+
```

Comment:
please consider using one section in case no difference for `linux` and `macos` as
`#elif defined(__linux__) || defined(__APPLE__) `